### PR TITLE
Slide binding. It now accepts onChange

### PR DESCRIPTION
**Pull request by DanMMX**
4 mayor changes:

* onClick acceptance through latest SimpleSlider dist files.
* slider double binding so any outer scope can access directly to the SimpleSlider API
* slider's slide double binding so more natural angular digest happens
* directive now waits until dynamic children are loaded

### CODE REVIEW

File: app/views/main.html [hunk 1]
```diff
@@ -32,7 +32,7 @@
   <option value="3">Slide 4</option>
 </select>
 
-<simple-slider style="width:612px; height:612px" auto-play="false" change="{{actualslide}}">
+<simple-slider simple-slider-slide="actualslide" style="width:612px; height:612px" auto-play="false">
```
Comment (ruyadorno) [inline app/views/main.html]: I would like to see a better attribute name for the slide attribute, `actual-slide` or `current-slide` are definitely better

File: Gruntfile.js [hunk 2]
```diff
@@ -334,7 +334,8 @@ module.exports = function (grunt) {
           'app/scripts/services/simplesliderservice.js',
           'app/scripts/directives/simple-slider.js'
         ],
-        dest: 'dist/angular-simple-slider.js' },
+        dest: 'dist/angular-simple-slider.js'
+      },
```
Comment (ruyadorno) [inline Gruntfile.js]: Unnecessary change here :)

Comment (DanMMX) [inline Gruntfile.js]: jshint complains about closing brace in the same line, do you still want me to revert that?

**Comment by ruyadorno** [comment]
Many thanks @DanMMX ! :smile: 

These are awesome ideas and fixes! I'm taking the time to comment on a few things and let's get it merged :)

**Comment by DanMMX** [comment]
Good to know @ruyadorno ^_^ let me know when you finished commenting so we don't overlap changes + comments.

File: Gruntfile.js (hunk 2 above)
Comment (ruyadorno) [inline Gruntfile.js]: hmmm, mine doesn't complain, maybe I need to specify some missing setting on `.jshintrc`.

Anyways, it's not related to the changes so it's better to revert it for the sake of keeping the PR focused only on the things it's meant to fix :)

File: app/scripts/directives/simple-slider.js [hunk 3]
```diff
@@ -1,20 +1,50 @@
...
 
     return {
 
-      restrict: 'E',
-      scope: {},
+      restrict: 'AE',
+      scope: {
+        onChange: '&',
+        slide: '=?simpleSliderSlide',
+        slider: '=?simpleSlider'
```
Comment (ruyadorno) [inline app/scripts/directives/simple-slider.js]: I have to confess that my angular skills are a bit rusty these days, so help me out with this one.

Is it really a good idea to expose the `SimpleSlider` object as a `simple-slider` attribute?

My original idea was to hide everything behind the directive implementation and the consumer would just play with the attributes. I don't really know what are the trade-offs involved here. Since you're already exposing `onchange` and the `simple-slider-slide` attribute it seems that it would make sense to keep the consumers away from the simpleSlider instance, have you give it some thought? I would love to have some feedback :blush:

File: app/views/main.html [hunk 4]
```diff
@@ -32,11 +32,14 @@
   <option value="3">Slide 4</option>
 </select>
 
-<simple-slider style="width:612px; height:612px" auto-play="false" change="{{actualslide}}">
+<button ng-click="slider.prev()">Prev Slide</button>
+<button ng-click="slider.next()">Next Slide</button>
```
Comment (ruyadorno) [inline app/views/main.html]: :+1: the example surely helps to show the intention of having the `slider-instance` attribute there :smile:

File: app/scripts/directives/simple-slider.js [hunk 5]
```diff
@@ -1,20 +1,50 @@
...
-      scope: {},
+      restrict: 'AE',
+      scope: {
+        onChange: '&',
+        current: '=?currentSlide',
+        slider: '=?sliderInstance'
+      },
 
       link: function postLink(scope, element, attrs) {
+        var options = attrs, watcher;
```
Comment (ruyadorno) [inline app/scripts/directives/simple-slider.js]: maybe rename `watcher` to a more informative name? I find `disposeWatcher` to be a better name in this case, I had to go to the angular documentation to remember what `$scope.watch` would return :stuck_out_tongue_closed_eyes:

**Comment by ruyadorno** [comment]
Looking awesome :smile: I'm done with reviews, just the `watcher` var renaming and we're good to go

File: app/views/main.html (hunk 4 above)
Comment (DanMMX) [inline app/views/main.html]: Glad it was self explaining, if you want further uses let me know, I used it on a test for some job. 

Also added a demo with dynamic images :wink:

**Comment by DanMMX** [comment]
I think it's ready @ruyadorno :+1:

**Comment by ruyadorno** [comment]
:+1: LGTM
